spec/features/global_chat_spec.rb

```diff
@@ -35,17 +30,29 @@ def post_message(message)
     end
   end
 
-  it 'shows the correct timestamp' do
-    post_message(message)
-    message = RoomMessage.all.last
-    expect(page).to have_text(message.formatted_time)
-  end
+  context 'with a posted message' do
+    let(:room_message) { RoomMessage.all.last }
+    before do
+      post_message(message)
+      room_message
+    end
 
-  it 'shows the correct timestamp after reload' do
-    post_message(message)
-    message = RoomMessage.all.last
-    visit root_path
-    expect(page).to have_text(message.formatted_time)
+    it 'clears text box after message submit' do
+      expect(page).to have_field('room_message_message', text: '')
+    end
+
+    it 'shows the correct timestamp' do
+      expect(page).to have_text(room_message.formatted_time)
+    end
+
+    it "displays the user's avatar" do
+      expect(page).to have_css("img[src='#{avatar_user_path(user)}']")
+    end
+
+    it 'shows the correct timestamp after reload' do
```

Comment:
I would check for all things (message, time, avatar, avatar link, name and name link) after posting and after reloading. Should be easy using shared examples and only a different before block. Can also be moved to another PR if you would prefer that